Benchmarks/BenchmarkReporter.swift

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Protocol that defines how are the benchmark results printed to the stdout. 
+protocol BenchmarkReporter {
+
+    /// Print benchmark results to stdout.
+    func printResults(_ result: BenchmarkResults)
+
+    /// Print benchmark default settings to stdout.
+    func printDefaults()
+}
+
+/// Reporter that prints results as human-readable plain text.
+struct PlainTextReporter: BenchmarkReporter {
+
+    func printResults(_ result: BenchmarkResults) {
+        let (average, standardDeviation) = statistics(for: result.interpretedTimings)
+
+        switch result.variety {
+        case .inferenceThroughput:
+            print("Benchmark: \(result.name):")
+            print("\tAfter \(result.settings.iterations) iterations:")
+            print(
+                "\tSamples per second: \(String(format: "%.2f", average)), standard deviation: \(String(format: "%.2f", standardDeviation))"
+            )
+        case .trainingTime:
+            print("Benchmark: \(result.name):")
+            print("\tAfter \(result.settings.iterations) iterations:")
+            print(
+                "\tAverage: \(String(format: "%.2f", average)) ms, standard deviation: \(String(format: "%.2f", standardDeviation)) ms"
+            )
+        }
+    }
+
+    func printDefaults() {
+        for (name, benchmarkModel) in benchmarkModels {
+            let trainingFlags = flags(from: benchmarkModel.defaultTrainingSettings)
+            print("--benchmark \(name) --training \(trainingFlags)")
+            let inferenceFlags = flags(from: benchmarkModel.defaultInferenceSettings)
+            print("--benchmark \(name) --inference \(inferenceFlags)")
+        }
+    }
+}
+
+/// Auxiliary struct used to show benchmark default settings coupled with benchmark name.
+struct JsonDefaults: Codable {
```

Comment:
This struct doesn't seem tied to JSON and doesn't seem to have inherent default-ness. How about renaming it to `BenchmarkConfiguration`?
```suggestion
struct BenchmarkConfiguration: Codable {
```

Reply:
Yep, good call, fixed in 5bc569a.